layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php

```diff
@@ -23,19 +26,58 @@
     \update_option('graphql-api-extension', true);
 });
 
-// If the GraphQL API plugin is active => Create and set-up the extension
+/**
+ * Create and set-up the extension
+ */
 add_action('plugins_loaded', function (): void {
+    /**
+     * Validate the GraphQL API plugin is active
+     */
     if (!class_exists('\GraphQLAPI\GraphQLAPI\Plugin')) {
+        \add_action('admin_notices', function () {
+            _e(sprintf(
+                '<div class="notice notice-error">' .
+                    '<p>%s</p>' .
+                '</div>',
+                sprintf(
+                    __('Plugin <strong>%s</strong> is not installed or activated. Without it, plugin <strong>%s</strong> will not be loaded.', 'graphql-api-events-manager'),
+                    __('GraphQL API for WordPress', 'graphql-api-convert-case-directives'),
+                    __('GraphQL API - Convert Case Directives', 'graphql-api-convert-case-directives')
+                )
+            ));
+        });
         return;
     }
 
-    define('GRAPHQL_API_CONVERT_CASE_DIRECTIVES_PLUGIN_FILE', __FILE__);
-    define('GRAPHQL_API_CONVERT_CASE_DIRECTIVES_VERSION', '0.7.13');
-    define('GRAPHQL_API_CONVERT_CASE_DIRECTIVES_DIR', dirname(__FILE__));
-    define('GRAPHQL_API_CONVERT_CASE_DIRECTIVES_URL', plugin_dir_url(__FILE__));
+    /**
+     * Make sure this plugin is not duplicated.
+     */
+    if (class_exists('\GraphQLAPI\ConvertCaseDirectives\PluginInfo')) {
+        \add_action('admin_notices', function () {
+            _e(sprintf(
+                '<div class="notice notice-error">' .
+                    '<p>%s</p>' .
+                '</div>',
+                sprintf(
+                    __('Plugin <strong>%s</strong> is already installed with version <code>%s</code>, so version <code>%s</code> has not been loaded. Please deactivate all versions, remove the older version, and activate again the latest version of the plugin.', 'graphql-api'),
+                    __('GraphQL API - Convert Case Directives', 'graphql-api-events-manager'),
+                    PluginInfo::get('version'),
+                    '0.7.13'
+                )
+            ));
+        });
+        return;
+    }
 
     // Load Composer’s autoloader
     require_once(__DIR__ . '/vendor/autoload.php');
 
-    (new \GraphQLAPI\ConvertCaseDirectives\GraphQLAPIExtension(__FILE__))->setup();
+    // Initialize the Plugin information
+    PluginInfo::init([
+        'version' => '0.7.13',
+        'dir' => dirname(__FILE__),
+        'url' => plugin_dir_url(__FILE__),
```

Comment:
@leoloso Storing `plugin_dir_url()` out at any time prevents CDN - and other - plugins from working 😢

Reply:
Good point! But that is just for internal use, to load the images for the documentation, displayed in the wp-admin, so it should work fine. (Also that's just for DEV, for PROD the images are excluded from the plugin, and retrieved from GitHub)